doctests regression with v3.7.3 on OS-X

- [x] Include a detailed description of the bug or suggestion

After upgrading to v3.7.3, all doctests started failing with the exception:

```
<my-pyenv>/lib/python3.7/site-packages/_pytest/doctest.py:206: TypeError: cannot unpack non-iterable NoneType object
```

Everything is OK with v3.7.2.

Slightly longer traceback (For the record, looking into the stacktrace, I see some references to "darwin"-specific functions):

```
___________________________________________________________________ [doctest] xxx ___________________________________________________________________

self = <CallInfo when='call' exception: cannot unpack non-iterable NoneType object>, func = <function call_runtest_hook.<locals>.<lambda> at 0x1112f49d8>, when = 'call'
treat_keyboard_interrupt_as_exception = False

    def __init__(self, func, when, treat_keyboard_interrupt_as_exception=False):
        #: context of invocation: one of "setup", "call",
        #: "teardown", "memocollect"
        self.when = when
        self.start = time()
        try:
>           self.result = func()

<my-pyenv>/lib/python3.7/site-packages/_pytest/runner.py:201:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

>   lambda: ihook(item=item, **kwds),
    when=when,
    treat_keyboard_interrupt_as_exception=item.config.getvalue("usepdb"),
        )

<my-pyenv>/lib/python3.7/site-packages/_pytest/runner.py:183:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <_HookCaller 'pytest_runtest_call'>, args = (), kwargs = {'item': <DoctestItem 'somemodule.somefunction'>}, notincall = set()

    def __call__(self, *args, **kwargs):
        if args:
            raise TypeError("hook calling supports only keyword arguments")
        assert not self.is_historic()
        if self.argnames:
            notincall = set(self.argnames) - set(['__multicall__']) - set(
                kwargs.keys())
            if notincall:
                warnings.warn(
                    "Argument(s) {} which are declared in the hookspec "
                    "can not be found in this hook call"
                    .format(tuple(notincall)),
                    stacklevel=2,
                )
>       return self._hookexec(self, self._nonwrappers + self._wrappers, kwargs)

<my-pyenv>/lib/python3.7/site-packages/pluggy/hooks.py:258:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <_pytest.config.PytestPluginManager object at 0x1034317b8>, hook = <_HookCaller 'pytest_runtest_call'>
methods = [<HookImpl plugin_name='runner', plugin=<module '_pytest.runner' from '<my-pyenv......t 0x103d4f4e0>>, <HookImpl plugin_name='logging-plugin', plugin=<_pytest.logging.LoggingPlugin object at 0x10e8f9860>>]
kwargs = {'item': <DoctestItem 'somemodule.somefunction'>}

    def _hookexec(self, hook, methods, kwargs):
        # called from all hookcaller instances.
        # enable_tracing will set its own wrapping function at self._inner_hookexec
>       return self._inner_hookexec(hook, methods, kwargs)

<my-pyenv>/lib/python3.7/site-packages/pluggy/manager.py:67:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

hook = <_HookCaller 'pytest_runtest_call'>
methods = [<HookImpl plugin_name='runner', plugin=<module '_pytest.runner' from '<my-pyenv......t 0x103d4f4e0>>, <HookImpl plugin_name='logging-plugin', plugin=<_pytest.logging.LoggingPlugin object at 0x10e8f9860>>]
kwargs = {'item': <DoctestItem 'somemodule.somefunction'>}

    self._inner_hookexec = lambda hook, methods, kwargs: \
        hook.multicall(
            methods, kwargs,
>           firstresult=hook.spec_opts.get('firstresult'),
        )

<my-pyenv>/lib/python3.7/site-packages/pluggy/manager.py:61:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

hook_impls = [<HookImpl plugin_name='runner', plugin=<module '_pytest.runner' from '<my-pyenv......t 0x103d4f4e0>>, <HookImpl plugin_name='logging-plugin', plugin=<_pytest.logging.LoggingPlugin object at 0x10e8f9860>>]
caller_kwargs = {'item': <DoctestItem 'somemodule.somefunction'>}, firstresult = False

    def _multicall(hook_impls, caller_kwargs, firstresult=False):
        """Execute a call into multiple python functions/methods and return the
        result(s).

        ``caller_kwargs`` comes from _HookCaller.__call__().
        """
        __tracebackhide__ = True
        results = []
        excinfo = None
        try:  # run impl and wrapper setup functions in a loop
            teardowns = []
            try:
                for hook_impl in reversed(hook_impls):
                    try:
                        args = [caller_kwargs[argname] for argname in hook_impl.argnames]
                    except KeyError:
                        for argname in hook_impl.argnames:
                            if argname not in caller_kwargs:
                                raise HookCallError(
                                    "hook call must provide argument %r" % (argname,))

                    if hook_impl.hookwrapper:
                        try:
                            gen = hook_impl.function(*args)
                            next(gen)   # first yield
                            teardowns.append(gen)
                        except StopIteration:
                            _raise_wrapfail(gen, "did not yield")
                    else:
                        res = hook_impl.function(*args)
                        if res is not None:
                            results.append(res)
                            if firstresult:  # halt further impl calls
                                break
            except BaseException:
                excinfo = sys.exc_info()
        finally:
            if firstresult:  # first result hooks return a single value
                outcome = _Result(results[0] if results else None, excinfo)
            else:
                outcome = _Result(results, excinfo)

            # run all wrapper post-yield blocks
            for gen in reversed(teardowns):
                try:
                    gen.send(outcome)
                    _raise_wrapfail(gen, "has second yield")
                except StopIteration:
                    pass

>           return outcome.get_result()

<my-pyenv>/lib/python3.7/site-packages/pluggy/callers.py:201:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <pluggy.callers._Result object at 0x11116efd0>

    def get_result(self):
        """Get the result(s) for this hook call.

            If the hook was marked as a ``firstresult`` only a single value
            will be returned otherwise a list of results.
            """
        __tracebackhide__ = True
        if self._excinfo is None:
            return self._result
        else:
            ex = self._excinfo
            if _py3:
>               raise ex[1].with_traceback(ex[2])

<my-pyenv>/lib/python3.7/site-packages/pluggy/callers.py:76:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

hook_impls = [<HookImpl plugin_name='runner', plugin=<module '_pytest.runner' from '<my-pyenv......t 0x103d4f4e0>>, <HookImpl plugin_name='logging-plugin', plugin=<_pytest.logging.LoggingPlugin object at 0x10e8f9860>>]
caller_kwargs = {'item': <DoctestItem 'somemodule.somefunction'>}, firstresult = False

    def _multicall(hook_impls, caller_kwargs, firstresult=False):
        """Execute a call into multiple python functions/methods and return the
        result(s).

        ``caller_kwargs`` comes from _HookCaller.__call__().
        """
        __tracebackhide__ = True
        results = []
        excinfo = None
        try:  # run impl and wrapper setup functions in a loop
            teardowns = []
            try:
                for hook_impl in reversed(hook_impls):
                    try:
                        args = [caller_kwargs[argname] for argname in hook_impl.argnames]
                    except KeyError:
                        for argname in hook_impl.argnames:
                            if argname not in caller_kwargs:
                                raise HookCallError(
                                    "hook call must provide argument %r" % (argname,))

                    if hook_impl.hookwrapper:
                        try:
                            gen = hook_impl.function(*args)
                            next(gen)   # first yield
                            teardowns.append(gen)
                        except StopIteration:
                            _raise_wrapfail(gen, "did not yield")
                    else:
>                       res = hook_impl.function(*args)

<my-pyenv>/lib/python3.7/site-packages/pluggy/callers.py:180:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

item = <DoctestItem 'somemodule.somefunction'>

    def pytest_runtest_call(item):
        _update_current_test_var(item, "call")
        sys.last_type, sys.last_value, sys.last_traceback = (None, None, None)
        try:
>           item.runtest()

<my-pyenv>/lib/python3.7/site-packages/_pytest/runner.py:111:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <DoctestItem 'somemodule.somefunction'>

    def runtest(self):
        _check_all_skipped(self.dtest)
>       self._disable_output_capturing_for_darwin()

<my-pyenv>/lib/python3.7/site-packages/_pytest/doctest.py:192:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <DoctestItem 'somemodule.somefunction'>

    def _disable_output_capturing_for_darwin(self):
        """
            Disable output capturing. Otherwise, stdout is lost to doctest (#985)
            """
        if platform.system() != "Darwin":
            return
        capman = self.config.pluginmanager.getplugin("capturemanager")
        if capman:
>           out, err = capman.suspend_global_capture(in_=True)
E           TypeError: cannot unpack non-iterable NoneType object

<my-pyenv>/lib/python3.7/site-packages/_pytest/doctest.py:206: TypeError
```

- [x] `pip list` of the virtual environment you are using

```
...
mypy                     0.620
pytest                   3.7.3
pytest-cov               2.5.1
pytest-django            3.4.2
tox                      3.2.1
...
```

- [x] pytest and operating system versions

`pytest v3.7.3` and 

```
$ sw_vers
ProductName:	Mac OS X
ProductVersion:	10.13.6
BuildVersion:	17G65
```

- [ ] Minimal example if possible

--- a/.travis.yml
+++ b/.travis.yml
@@ -54,8 +54,6 @@ jobs:
     - <<: *test-macos
       env: TOXENV=py27
     - <<: *test-macos
-      env: TOXENV=py36
-    - <<: *test-macos
       env: TOXENV=py37
       before_install:
         - brew update

--- a/.travis.yml
+++ b/.travis.yml
@@ -49,13 +49,12 @@ jobs:
       os: osx
       osx_image: xcode9.4
       sudo: required
-      env: TOXENV=py27
       install:
-        - python -m pip install --upgrade --pre tox
+        - python -m pip install --pre tox
+    - <<: *test-macos
+      env: TOXENV=py27
     - <<: *test-macos
       env: TOXENV=py36
-      install:
-        - python3 -m pip install --upgrade --pre tox
     - <<: *test-macos
       env: TOXENV=py37
       before_install:

--- a/.travis.yml
+++ b/.travis.yml
@@ -51,7 +51,6 @@ jobs:
       sudo: required
       install:
         - python -m pip install --pre tox
-    - <<: *test-macos
       env: TOXENV=py27
     - <<: *test-macos
       env: TOXENV=py37

--- a/.travis.yml
+++ b/.travis.yml
@@ -44,6 +44,25 @@ jobs:
       python: '3.7'
       sudo: required
       dist: xenial
+    - &test-macos
+      language: generic
+      os: osx
+      osx_image: xcode9.4
+      sudo: required
+      env: TOXENV=py27
+      install:
+        - python -m pip install --upgrade --pre tox
+    - <<: *test-macos
+      env: TOXENV=py36
+      install:
+        - python3 -m pip install --upgrade --pre tox
+    - <<: *test-macos
+      env: TOXENV=py37
+      before_install:
+        - brew update
+        - brew upgrade python
+        - brew unlink python
+        - brew link python
 
     - stage: deploy
       python: '3.6'

--- a/changelog/3888.bugfix.rst
+++ b/changelog/3888.bugfix.rst
@@ -1,0 +1,1 @@
+Fix macOS specific code using ``capturemanager`` plugin in doctests.

--- a/src/_pytest/doctest.py
+++ b/src/_pytest/doctest.py
@@ -203,7 +203,8 @@ class DoctestItem(pytest.Item):
             return
         capman = self.config.pluginmanager.getplugin("capturemanager")
         if capman:
-            out, err = capman.suspend_global_capture(in_=True)
+            capman.suspend_global_capture(in_=True)
+            out, err = capman.read_global_capture()
             sys.stdout.write(out)
             sys.stderr.write(err)
 

--- a/testing/python/fixture.py
+++ b/testing/python/fixture.py
@@ -1584,6 +1584,7 @@ class TestFixtureManagerParseFactories(object):
             values = []
         """
         )
+        testdir.syspathinsert(testdir.tmpdir.dirname)
         package = testdir.mkdir("package")
         package.join("__init__.py").write("")
         package.join("conftest.py").write(

